### ISSUE TITLE
wandb init hasn't been called

### ISSUE THREAD

**Issue by Ierezell** [wandb/client#557]
* Weights and Biases version: 0.8.10
* Python version: 3.7.4
* Operating System: Windows

### Description
Trying to launch training code in pytorch.
Launching code as recommended with` wandb run python train.py`
I have a` wandb.init(...)` at the begining of my code.

This code works perfectly fine on linux.

### What I Did
Put all my code in a` if __name__ == "__main__"` because windows needs it to launch processes : No success

```
wandb run python train.py 

wandb: ERROR wandb.init hasn't been called, can't configure run
Traceback (most recent call last):
  File "train.py", line 164, in <module>
    wandb.save(PATH_WEIGHTS_EMBEDDER)
  File "C:\Users\pisne\AppData\Local\Continuum\anaconda3\lib\site-packages\wandb\__init__.py", line 493, in save
    os.symlink(abs_path, wandb_path)
OSError: symbolic link privilege not held
wandb: Program failed with code 1. Press ctrl-c to abort syncing.
```

**Comment by vanpelt** [comment]
This looks like the real error is from os privileges.  It tried to create a symlink and got permission denied.  Can you try running it from an admin console?

**Comment by alberduris** [comment]
Same problem here. I can only configure the run with the command:

```
wandb init 
```
but then you can only set the `username` and `project-name`.

No matter what, when the model starts running always throw the warning that _"wandb init hasn't been called"_ which makes impossible to correctly set up a run. By the way, the rest of the features runs perfectly fine.

**Comment by Ierezell** [comment]
@vanpelt Tried in an admin console and get the same result. 

**Update** : 

- With `wandb.save(Path_to_my_weights)` i'm running into this issue. 

- Without `wandb.save(Path_to_my_weights)` the code runs but the init is not correctly done (the name and the id are set by default with fancy name but not mine) at least I get scalar logged. 
During training I receive the message `wandb: ERROR wandb.init hasn't been called, can't configure run`  (but the wandb init is done at the top of my script)
At least it runs.... 

With `wandb.save(Hard_coded_window_path)`  idem than without wandb.save(Path_to_my_weights)

**Comment by raubitsj** [comment]
@Ierezell 
I just reproduced your issue with wandb.save() on a windows VM.
The message is coming from wand.save() as you noticed and it is something we will fix in a later release.

If you want to configure your run parameters on windows, please use the environment variables or the command line.  for setting the name of your run you can set:
WANDB_NAME=
or you can run:
wandb run --name=myname python train.py

To fix the symlink issue, you can add symlink permissions to your user in windows:
Security Settings → Local Policies → User Rights Assignment → Create symbolic links

**Comment by alberduris** [comment]
Hey @raubitsj,

A couple of questions;

1. With the command line, the options are very limited, to the extent that you can not configure the project, which is vital.

```
Usage: wandb run [OPTIONS] PROGRAM [ARGS]...

  Launch a job

Options:
  --id TEXT                    Run id to use, default is to generate.
  --resume [never|must|allow]  Resume strategy, default is never
  --dir TEXT                   Files in this directory will be saved to wandb,
                               defaults to wandb
  --configs TEXT               Config file paths to load
  --name TEXT                  Name of the run, default is auto generated.
  --notes TEXT                 Notes to associate with the run.
  --show / --no-show           Open the run page in your default browser.
  --tags TEXT                  Tags to associate with the run (comma
                               seperated).
  --run_group TEXT             Run group to associate with the run.
  --job_type TEXT              Job type to associate with the run.
  --help                       Show this message and exit.
```
2. The environment variables seem to have no effect. 

I am trying to run the `tutorial.py` script on Windows and storing it under a custom project. To that end, and following your instructions, I have updated these lines of code:

```
#Initialize wandb
os.environ['WANDB_PROJECT'] = 'myproject'  # Use environment variables
wandb.init() 
config = wandb.config
```

And then I get

```
wandb: Local directory: wandb\run-20190919_074801-ofuaplgn
wandb: Syncing run jolly-water-13: https://app.wandb.ai/4lb3rduris/**uncategorized**/runs/ofuaplgn
wandb: Run `wandb off` to turn off syncing.
```

So as you can see, the project is not set correctly even after setting the value of the environment variable manually. By the way, it seems that you have to call the wandb.init() even though it throws a warning, because if you don't do it, new errors arises...

**Comment by raubitsj** [comment]
Thank you for pointing out that project is missing from the command line.  I think the reasoning is that you can set the project name with the `wandb init` command before running `wandb run`

The environment variables should work but you need to set them before you run `wandb run` as this is when the run is actually created.

It is unfortunate that the windows workflow is different but the way windows handles processes and output was so different it was difficult to keep them the same as the unix based systems.  It probably makes sense for us to add a section about windows usage on the docs.wandb.com site which explains this.

Thanks again for pointing these issues out, let us know if the combination of using setting the project (with either `wandb init` or setting environment variables) and other params with `wandb run` still has issues.

**Comment by alberduris** [comment]
Thank you for your explanations.

Unfortunately, the constraints of the Windows version make the product almost useless for me, and that's a shame, as you have a really great software. 

This is about saving time while experimenting, so automation is a must. I'm afraid that if I have to i) Set the project with `wandb init` (which opens a user prompt) and ii) Change the environment variables before the execution of the script, then the automation of launching experiments it's impossible. 

Sometimes we have to launch hundreds of runs from different experiments in sequence and let them run, clearly we can't be handling a user prompt plus variable assignment outside script for each run. 

In summary, everything should be doable with code inside the script, to be able to parametrize the runs via script parameters. If at some point in time that could be done on Windows, I would be grateful to know and try Wandb again!

**Comment by Ierezell** [comment]
@alberduris I know it's only a patch but you can use `Popen` from subprocesses in python to automate launching commands with arguments...

@raubitsj Thanks a lot for the precisions. I'm a linux power user and the unix wandb is amazing but unfornutaltely my lab granted me acces to a really powerfull Gpu.... on windows.....

I wanted the name of my run to be linked to parameters (like Arch_8_0.0001 etc...) which is good under linux but I will try on windows with `os.environ` and tell here my successes.

Thanks again for the fast and precise feedback !

**Comment by raubitsj** [comment]
We now support windows with the same interface as Linux/Mac.

You can start your script with python and use all the parameters to wandb.init() 

Marking closed. Please reopen if you have any issues with the improved windows support.

**Comment by noklam** [comment]
I hit the same issue with wandb==0.9.2 and wandb==0.9.4, tried run as admin as well.

**Comment by raubitsj** [comment]
Please do not use `wandb run` any more (This command will likely be removed in future releases)

Example python:
```
import wandb
def train():
    wandb.init()
    wandb.log(dict(metric=value)
if __name__ == "__main__":
  train()
```

Then run with:
`python train.py`

**Comment by noklam** [comment]
> @noklam Please do not use `wandb run` any more (This command will likely be removed in future releases)
> 
> Example python:
> 
> ```
> import wandb
> def train():
>     wandb.init()
>     wandb.log(dict(metric=value))
> if __name__ == "__main__":
>   train()
> ```
> 
> Then run with:
> `python train.py`

ops sorry, I came in this issue as I hit the os.symbolic link issue on Window.

in my case, it is due to wandb.save()

**Comment by rinnegann** [comment]
This link has worked for me https://www.scivision.dev/windows-symbolic-link-permission-enable/ , if you don't have  gpedit.msc in windows home edition you can download it from here https://www.itechtics.com/?dl_id=43
